### ISSUE TITLE
Error while deploying glusterfs on OpenShift Origin 3.9

### ISSUE THREAD

**Issue by ghost** [openshift/openshift-ansible#8869]
#### Trying to install GlusterFS on OpenShift Origin 3.9

while I am deploying using 
```
ansible-playbook -i host_glusterfs playbooks/deploy_cluster.yml -vvv --ask-become-pass
```
> It is throwing below error
```
"\tAllowing block volumes on cluster.",
"\tCreating node 192.168.68.26 ... Unable to create node: New Node doesn't have glusterd running",
"\tCreating node 192.168.68.22 ... Unable to create node: New Node doesn't have glusterd running",
"\tCreating node 192.168.68.51 ... Unable to create node: New Node doesn't have glusterd running"
```
> If i am deploying glusterfs standalone onto an existing cluster, it is adding only one node and getting
Error: failed to allocate new volume
```
 ansible-playbook -i host_glusterfs playbooks/openshift-glusterfs/config.yml -vvv --ask-become-pass
```
```
 "invocation": {
        "module_args": {
            "_raw_params": "oc --config=/tmp/openshift-glusterfs-ansible-dmq7hS/admin.kubeconfig rsh --namespace=glusterfs deploy-heketi-storage-1-h26ln heketi-cli -s http://localhost:8080 --user admin --secret 'UeO/FM48wmhwkcegGyJEm3A0SdxwUFbKjYomZEsok0I=' topology load --json=/tmp/openshift-glusterfs-ansible-dmq7hS/topology.json 2>&1",
            "_uses_shell": false,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": true
        }
    },
    "rc": 0,
    "start": "2018-06-20 13:32:05.895574",
    "stderr": "",
    "stderr_lines": [],
 "stdout": "Creating cluster ... ID: 6c19229a83918d2845964d07e7855364\n\tAllowing file volumes on cluster.\n\tAllowing block volumes on cluster.\n\tCreating node 192.168.68.26 ... ID: 894c617e0251b88d54841999a930abb8\n\t\tAdding device /dev/vdb ... ",
    "stdout_lines": [
        "Creating cluster ... ID: 6c19229a83918d2845964d07e7855364",
        "\tAllowing file volumes on cluster.",
        "\tAllowing block volumes on cluster.",
        "\tCreating node 192.168.68.26 ... ID: 894c617e0251b88d54841999a930abb8",
        "\t\tAdding device /dev/vdb ... "
    ]

```
```
"invocation": {
        "module_args": {
            "_raw_params": "oc --config=/tmp/openshift-glusterfs-ansible-dmq7hS/admin.kubeconfig rsh --namespace=glusterfs deploy-heketi-storage-1-h26ln heketi-cli -s http://localhost:8080 --user admin --secret 'UeO/FM48wmhwkcegGyJEm3A0SdxwUFbKjYomZEsok0I=' setup-openshift-heketi-storage --image heketi/heketi:latest --listfile /tmp/heketi-storage.json",
            "_uses_shell": false,
            "chdir": null,
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": true
        }
    },
    "msg": "non-zero return code",
    "rc": 255,
    "start": "2018-06-20 13:37:09.216300",
    "stderr": "Error: Failed to allocate new volume: No space\ncommand terminated with exit code 255",
    "stderr_lines": [
        "Error: Failed to allocate new volume: No space",
        "command terminated with exit code 255"
    ],
    "stdout": "",
    "stdout_lines": []
```
##### Version
Please put the following version information in the code block
indicated below.
* Your ansible version per `ansible --version`
`ansible 2.5.2`

If you're operating from a **git clone**:

* The output of `git describe`
`openshift-ansible-3.9.32-1-21-ge927493`

* Your operating system and version,`Red Hat Enterprise Linux Server release 7.5 (Maipo)`
*Inventory file
```
[OSEv3:children]
masters
nodes
etcd
lb
glusterfs
glusterfs_registry

[OSEv3:vars]
ansible_ssh_user=balanna
ansible_become=true
openshift_release='v3.9'
openshift_deployment_type=origin


os_firewall_use_firewalld=false
openshift_use_openshift_sdn=False
os_sdn_network_plugin_name='cni'
openshift_use_calico=True

openshift_master_cluster_method=native
openshift_clock_enabled=true
openshift_use_dnsmasq=true
openshift_enable_service_catalog=false
openshift_disable_check=memory_availability,disk_availability

openshift_master_default_subdomain=nssopenshift.com
openshift_master_cluster_hostname=nssopenshift.com
openshift_master_cluster_public_hostname=nssopenshift.com
openshift_public_hostname=nssopenshift.com

osm_use_cockpit=true
osm_cockpit_plugins=['cockpit-kubernetes']


openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_master_htpasswd_users={'admin': '$apr1$zgSjCrLt$1KSuj66CggeWSv.D.BXOA1', 'user': '$apr1$.gw8w9i1$ln9bfTRiD6OwuNTG5LvW50'}

openshift_repos_enable_testing=True
openshift_disable_check=memory_availability,disk_availability,docker_storage

openshift_router_selector='region=infra'

#registry
openshift_registry_selector="region=infra"
openshift_hosted_registry_storage_kind=glusterfs

openshift_metrics_install_metrics=true
openshift_metrics_hawkular_nodeselector={"region":"infra"}
openshift_metrics_cassandra_nodeselector={"region":"infra"}
openshift_metrics_heapster_nodeselector={"region":"infra"}
osn_storage_plugin_deps=['ceph','glusterfs','iscsi']

openshift_logging_es_nodeselector={"region":"infra"}
openshift_logging_kibana_nodeselector={"region":"infra"}
openshift_logging_curator_nodeselector={"region":"infra"}

openshift_storage_glusterfs_registry_block_deploy=true
openshift_storage_glusterfs_registry_block_host_vol_size=50
openshift_storage_glusterfs_registry_block_storageclass=true
openshift_storage_glusterfs_registry_block_storageclass_default=true

openshift_storage_glusterfs_wipe=True
openshift_storageclass_default=false



# host group for masters
[masters]
njbbapida13z.com        openshift_public_hostname=njbbapida13z.com
njbbapida14z.com        openshift_public_hostname=njbbapida14z.com
njbbapida20z.com        openshift_public_hostname=njbbapida20z.com
#njbbapida3z.com        openshift_public_hostname=njbbapida3z.com

[etcd]
njbbapida13z.com        openshift_public_hostname=njbbapida13z.com
njbbapida14z.com        openshift_public_hostname=njbbapida14z.com
njbbapida20z.com        openshift_public_hostname=njbbapida20z.com
#njbbapida3z.com        openshift_public_hostname=njbbapida3z.com

[lb]
njbbapida19z.com        openshift_public_hostname=njbbapida19z.com

[nodes]
njbbapida13z.com        openshift_public_hostname=njbbapida13z.com      openshift_schedulable=True      openshift_node_labels="{'region': 'infra','zone': 'default'}"
njbbapida14z.com        openshift_public_hostname=njbbapida14z.com      openshift_schedulable=True      openshift_node_labels="{'region': 'primary','zone': 'default'}"
njbbapida20z.com        openshift_public_hostname=njbbapida20z.com      openshift_schedulable=True      openshift_node_labels="{'region': 'primary','zone': 'default'}"
njbbapida17z.com        openshift_public_hostname=njbbapida17z.com      openshift_schedulable=True      openshift_node_labels="{'region': 'primary','zone': 'default'}"
njbbapida18z.com        openshift_public_hostname=njbbapida18z.com      openshift_schedulable=True      openshift_node_labels="{'region': 'infra','zone': 'default'}"
njbbapida3z.com         openshift_public_hostname=njbbapida3z.com       openshift_schedulable=True      openshift_node_labels="{'region': 'infra','zone': 'default'}"

[glusterfs]
njbbapida17z.com     glusterfs_devices='[ "/dev/vdb" ]'
njbbapida14z.com     glusterfs_devices='[ "/dev/vdb" ]'
njbbapida20z.com     glusterfs_devices='[ "/dev/vdb" ]'

[glusterfs_registry]
njbbapida13z.com     glusterfs_devices='[ "/dev/vdb" ]'
njbbapida18z.com     glusterfs_devices='[ "/dev/vdb" ]'
njbbapida3z.com     glusterfs_devices='[ "/dev/vdb3" ]'
```

**Comment by naveenbdevops** [comment]
below error for `Unable to create node: New Node doesn't have glusterd running`
```
 oc logs po/deploy-heketi-storage-1-8tjcb -n glusterfs
Setting up heketi database
No database file found
stat: cannot stat '/var/lib/heketi/heketi.db': No such file or directory
Heketi v6.0.0-1-g47f41a8-release-6
[heketi] INFO 2018/06/20 17:27:09 Loaded kubernetes executor
[heketi] INFO 2018/06/20 17:27:10 Block: Auto Create Block Hosting Volume set to true
[heketi] INFO 2018/06/20 17:27:10 Block: New Block Hosting Volume size 100 GB
[heketi] INFO 2018/06/20 17:27:10 GlusterFS Application Loaded
Authorization loaded
Listening on port 8080
[negroni] Started GET /clusters
[negroni] Completed 200 OK in 520.201µs
[negroni] Started GET /clusters
[negroni] Completed 200 OK in 347.693µs
[negroni] Started POST /clusters
[negroni] Completed 201 Created in 12.123939ms
[negroni] Started POST /nodes
[cmdexec] INFO 2018/06/20 17:27:23 Check Glusterd service status in node 192.168.68.26
[kubeexec] ERROR 2018/06/20 17:27:53 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:296: Get https://172.30.0.1:443/api/v1/namespaces/glusterfs/pods?labelSelector=glusterfs-node: dial tcp 172.30.0.1:443: i/o timeout
[kubeexec] ERROR 2018/06/20 17:27:53 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:297: Failed to get list of pods
[cmdexec] ERROR 2018/06/20 17:27:53 /src/github.com/heketi/heketi/executors/cmdexec/peer.go:76: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:27:53 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:106: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:27:53 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:107: New Node doesn't have glusterd running
[negroni] Completed 400 Bad Request in 30.009974975s
[negroni] Started POST /nodes
[cmdexec] INFO 2018/06/20 17:27:53 Check Glusterd service status in node 192.168.68.22
[kubeexec] ERROR 2018/06/20 17:28:23 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:296: Get https://172.30.0.1:443/api/v1/namespaces/glusterfs/pods?labelSelector=glusterfs-node: dial tcp 172.30.0.1:443: i/o timeout
[kubeexec] ERROR 2018/06/20 17:28:23 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:297: Failed to get list of pods
[cmdexec] ERROR 2018/06/20 17:28:23 /src/github.com/heketi/heketi/executors/cmdexec/peer.go:76: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:28:23 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:106: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:28:23 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:107: New Node doesn't have glusterd running
[negroni] Completed 400 Bad Request in 30.018999483s
[negroni] Started POST /nodes
[cmdexec] INFO 2018/06/20 17:28:23 Check Glusterd service status in node 192.168.68.51
[kubeexec] ERROR 2018/06/20 17:28:53 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:296: Get https://172.30.0.1:443/api/v1/namespaces/glusterfs/pods?labelSelector=glusterfs-node: dial tcp 172.30.0.1:443: i/o timeout
[kubeexec] ERROR 2018/06/20 17:28:53 /src/github.com/heketi/heketi/executors/kubeexec/kubeexec.go:297: Failed to get list of pods
[cmdexec] ERROR 2018/06/20 17:28:53 /src/github.com/heketi/heketi/executors/cmdexec/peer.go:76: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:28:53 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:106: Failed to get list of pods
[heketi] ERROR 2018/06/20 17:28:53 /src/github.com/heketi/heketi/apps/glusterfs/app_node.go:107: New Node doesn't have glusterd running
[negroni] Completed 400 Bad Request in 30.013911904s
[negroni] Started GET /clusters/5884156aabfcd3542585dc14e3963a9d
[negroni] Completed 200 OK in 527.359µs
[negroni] Started DELETE /clusters/5884156aabfcd3542585dc14e3963a9d
[heketi] INFO 2018/06/20 17:28:53 Deleted cluster [5884156aabfcd3542585dc14e3963a9d]
[negroni] Completed 200 OK in 14.115641ms
```

**Comment by bilalx20** [comment]
You need to hack the openshift-ansible role a little bit. Discussed and resolved with #1375

**Comment by mtnbikenc** [comment]
Closing old issues.
/close
